docs/proposals/csi-deployment-crd.md

```diff
@@ -0,0 +1,258 @@
+# Proposal:  CSI deployment CRD
+
+Last updated: 05.03.20
+
+## Abstract
+
+To deploy CSI component with operator, it's necessary to provide CRD so operator can deploy CSI programmable using its configuration 
+
+## Background
+
+Currently CSI is deployed using helm charts and value
+
+## Proposal
+
+Example of CRD:
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: csi.dell.com
+  labels:
+    app.kubernetes.io/name: csi-operator
+spec:
+  group: csi.dell.com
+  names:
+    kind: CSI
+    listKind: CSIList
+    plural: csi
```

Comment:
Is it possible to limit number of CR in CSI CRD?

Reply:
I didn't find such information. It's seems like CRD specification doesn't contain special field for this case. I suppose we can only handle it in operator code